### PR TITLE
Hide Motor Poles when it is not necessary

### PR DESCRIPTION
**Pull request by Asizon**
Hide Motor Poles tab for more cleanning, only shows when Dshot Bidir or ESC_SENSOR are actived. A lot of thanks @McGiverGim for your help for this ;)

hidemotorpoles

### CODE REVIEW

File: src/js/tabs/configuration.js [hunk 1]
```diff
@@ -511,11 +511,21 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
                 self.analyticsChanges['BidirectionalDshot'] = newValue;
 
                 MOTOR_CONFIG.use_dshot_telemetry = value;
+
```
Comment (McGiverGim) [inline src/js/tabs/configuration.js]: Don't add empty lines if not needed in this PR.

File: src/js/tabs/configuration.js [hunk 2]
```diff
@@ -511,11 +511,21 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
...
         }
 
+        function hidemotorPoles() {
+            if ($("input[id='dshotBidir']").is(':checked') ||  $("input[name='ESC_SENSOR']").is(':checked')){
+                $('div.motorPoles').show();
+
+            } else {
+                $('div.motorPoles').hide();
+            }
+		}
```
Comment (McGiverGim) [inline src/js/tabs/configuration.js]: Caution with the indentation 😉

I suggest better something like:
```
let motorPolesVisible = $("input[id='dshotBidir']").is(':checked') ||  $("input[name='ESC_SENSOR']").is(':checked');
$('div.motorPoles').toggle(motorPolesVisible);
```
The number of conditions can grow in future versions and this is cleaner.

File: src/js/tabs/configuration.js [hunk 3]
```diff
@@ -542,13 +552,16 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
 
             $('div.checkboxDshotBidir').toggle(semver.gte(CONFIG.apiVersion, "1.42.0") && digitalProtocol);
             $('div.motorPoles').toggle(semver.gte(CONFIG.apiVersion, "1.42.0"));
+			
+            //trigger change dshotBidir and ESC_SENSOR to show/hide Motor Poles tab
+			$("input[id='dshotBidir']").change(hidemotorPoles).change();
+			$("input[name='ESC_SENSOR']").change(hidemotorPoles);
```
Comment (McGiverGim) [inline src/js/tabs/configuration.js]: Indentation again.

File: src/js/tabs/configuration.js [hunk 4]
```diff
@@ -511,11 +511,21 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
                 self.analyticsChanges['BidirectionalDshot'] = newValue;
 
                 MOTOR_CONFIG.use_dshot_telemetry = value;
+				
             });
 
             $('input[name="motorPoles"]').val(MOTOR_CONFIG.motor_poles);
         }
 
+        function hidemotorPoles() {
```
Comment (McGiverGim) [inline src/js/tabs/configuration.js]: To be camelCase must be hideMotorPoles()

File: src/js/tabs/configuration.js [hunk 5]
```diff
@@ -516,6 +516,11 @@ TABS.configuration.initialize = function (callback, scrollPosition) {
             $('input[name="motorPoles"]').val(MOTOR_CONFIG.motor_poles);
         }
 
+        function hideMotorPoles() {
+            let motorPolesVisible = $("input[id='dshotBidir']").is(':checked') ||  $("input[name='ESC_SENSOR']").is(':checked');
```
Comment (mikeller) [inline src/js/tabs/configuration.js]: Double whitespace. :space_invader:

Comment (Docteh) [reply src/js/tabs/configuration.js]: After the two pipes?

Comment (mikeller) [reply src/js/tabs/configuration.js]: Yeap.

**Comment by Asizon** [comment]
@mikeller done! withe space removed and rebased

**Pull request merged**